src/Rector/ClassMethod/AddGenericReturnTypeToRelationsRector.php

```diff
@@ -476,4 +490,13 @@ private function setShouldUseNewGenerics(): void
             $this->shouldUseNewGenerics = version_compare($reflectionClassConstant->getValue(), '11.15.0', '>=');
         }
     }
+
+    private function setShouldUsePivotGeneric(): void
+    {
+        $reflectionClassConstant = new ReflectionClassConstant($this->applicationClass, 'VERSION');
+
+        if (is_string($reflectionClassConstant->getValue())) {
+            $this->shouldUsePivotGeneric = version_compare($reflectionClassConstant->getValue(), '12.3.0', '>=');
+        }
+    }
```

Comment:
Useful, have thought previously something like this should be an analyzer class in the project to make it a one line resuable method.